tests/test_sim_result.py

```diff
@@ -36,6 +36,121 @@ def test_pickle(self):
+        # print(result.data)  # [0.0, 2.5, 5.0, 7.5, 10.0]
+        result.remove(0) # Test specified index
+        self.assertEqual(result.times, [1, 2, 3, 4])
+        self.assertEqual(result.data, [2.5, 5.0, 7.5, 10.0])
+
+        # result.remove(1) # Test specified index
+        # self.assertEqual(result.times, [2, 3, 4]) # Passes
+        # self.assertEqual(result.data, [2.5, 5.0, 7.5, 10.0]) # wrong behavior? ValueError 
+
+        # because ValueError, any type can be passed to be removed
+        # self.assertRaises(TypeError, result.remove, ) # Test no index specified
+        # self.assertRaises(TypeError, result.remove, "5") # Tests specifying an invalid index type
+        # self.assertRaises(TypeError, result.remove, [0,1])
+        # self.assertRaises(TypeError, result.remove, {})
+        # self.assertRaises(TypeError, result.remove, set())
+        # self.assertRaises(TypeError, result.remove, 1.5)
+
+        # self.assertRaises(IndexError, result.remove, 5) # Test specifying an invalid value
+        # self.assertRaises(IndexError, result.remove, 3)
+        # self.assertRaises(TypeError, result.remove, True) # This removes index 1 because boolean True, is this okay?
+
+    def test_clear(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+        self.assertEqual(result.times, [0, 1, 2, 3, 4])
+        self.assertEqual(result.data, [0.0, 2.5, 5.0, 7.5, 10.0])
+        self.assertRaises(TypeError, result.clear, True)
+
+        result.clear()
+        self.assertEqual(result.times, [])
+        self.assertEqual(result.data, [])
+
+    def test_time(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+        self.assertEqual(result.time(0), result.times[0])
+        self.assertEqual(result.time(1), result.times[1])
+        self.assertEqual(result.time(2), result.times[2])
+        self.assertEqual(result.time(3), result.times[3])
+        self.assertEqual(result.time(4), result.times[4])
+
+        self.assertRaises(TypeError, result.time, ) # Test no input given
+        self.assertRaises(TypeError, result.time, "0") # Tests specifying an invalid index type 
+        self.assertRaises(TypeError, result.time, [0,1])
+        self.assertRaises(TypeError, result.time, {})
+        self.assertRaises(TypeError, result.time, set())
+        self.assertRaises(TypeError, result.time, 1.5)
+        # self.assertRaises(TypeError, result.time, True) # Test no input given
+
+    def test_plot(self):
+        NUM_ELEMENTS = 5 # Creating two result objects
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+        # INCOMPLETE
```

Comment:
As we discussed, minimum test is just running the plot function and making sure it doesn't raise an exception. Something more would be nice, but not essential